protected/humhub/modules/content/controllers/ContentController.php

```diff
@@ -394,4 +396,25 @@ public function actionNotificationSwitch()
 
         return $this->asJson($json);
     }
+
+    public function actionScheduleOptions($id = null)
+    {
+        $this->forcePostRequest();
+
```

Comment:
Is there any protection? We also need to ensure this is only possible for New or Draft Contents

Reply:
@luke- Yes, it works well for New and Draft contents. I just decided to this this POST protection because in almost all cases we need to call this as POST request.
The action is loaded now only by modal window, and I decided to use js code `modal.post`(POST request) instead of `modal.load`(GET request).
Please review what cases we have:

- load a modal form first time for new created content: here it is possible to use GET
- submit the modal form: only POST request
- load the modal form next times for the same new content in order to do changes: here we must use only POST request, in order to load date and time from previous submitted form(need POST request for php code `ScheduleOptionsForm::load($_POST)`)
- load it for existing/stored content: possible to use GET request
- submit the changed options for the existing content: only POST request

i.e. in all cases we use the same js code which can use either `modal.load` or `modal.post` and it is easy to use in all cases only one way, so I selected to use `modal.post` in all cases and it is why I decide to use here the POST protection because I don't think the action will may be called in some cases as GET request.

Reply:
@yurabakhtin Thanks for the explaination. Ok, the GET/POST methods are ok. I was more concerned about whether the user should be allowed to view and edit the content. Or is that validated by the model?

Reply:
@yurabakhtin 

> Or is that validated by the model?

Do you mean to add some additional checking like `$content->canEdit()` for this action?

Reply:
@yurabakhtin Is there already any check, that the user can edit the draft or schedulded content?

Reply:
@luke- No yet. I started new method `Content::canChangeState($state)`, it is used only now in the method `Content::setState()`, and the method `canChangeState()` checks only for allowed state values, e.g. if wrong state value is passed from request.

I thought to extend this method later to check if current user is a content owner then allow to change state to serveral values like `draft`, `scheduled`, but I am not sure what states should be allowed and what denied depending on user permissions, could you advise me here?

Or may be we should integrate the state checking inside the existing method `Content::canEdit()`?

Reply:
@luke- Ok, done here https://github.com/humhub/humhub/pull/6116/commits/3f75a583b84335312330bd5f0c3ce56fe4101eb7#diff-3df0e8a05df0ffddaaac0a53aed588b00cdcbf899d0c1973caa3bd5d18247082R406.